src/functional/filter.go

```diff
@@ -0,0 +1,11 @@
+package functional
+
+func Filter[T any](arr []T, filterFn func(val T) bool) []T {
+	ret := make([]T, 0)
+	for _, val := range arr {
+		if filterFn(val) {
+			ret = append(ret, val)
+		}
+	}
+	return ret
+}
```

Comment:
```suggestion
func Filter[T any](array []T, predicate func(value T) bool) []T {
	filteredResult := make([]T, 0)
	
	for _, element := range array {
		if predicate(element) {
			filteredResult = append(filteredResult, element)
		}
	}
	
	return filteredResult
}
```

Reply:
Somehow I f'ed up the intendation, please disregard the w i d e tabs :sweat_smile: